src/live.rs

```diff
@@ -493,6 +542,30 @@ impl<'a> EmbedArea<'a> {
         Ok(())
     }
 
+    fn stream(&mut self, buf: &[u8], writer: &mut (impl Write + ?Sized)) -> Result<()> {
+        let mut buffer = [0u8; 1024 * 1024];
+        self.file
+            .seek(SeekFrom::Start(0))
+            .chain_err(|| "seeking to start")?;
+        copy_exactly_n(&mut self.file, writer, self.offset, &mut buffer)
+            .chain_err(|| "copying file")?;
+        if buf.len() > self.length {
+            bail!("buffer larger than embed area");
+        }
+        writer.write_all(buf).chain_err(|| "writing embed area")?;
+        let zeroes = vec![0; self.length - buf.len()];
+        writer.write_all(&zeroes).chain_err(|| "writing zeros")?;
+        self.file
+            .seek(SeekFrom::Start(self.offset + self.length as u64))
+            .chain_err(|| "seeking to end of embed area")?;
+        copy(
+            &mut BufReader::with_capacity(BUFFER_SIZE, self.file),
+            &mut BufWriter::with_capacity(BUFFER_SIZE, writer),
```

Comment:
The `BufWriter` [docs](https://doc.rust-lang.org/std/io/struct.BufWriter.html) recommend an explicit `flush()`.